### ISSUE TITLE
VibranceGUI affects secondary screen

### ISSUE THREAD

**Issue by DetectiveSquirrel** [juvlarN/vibranceGUI#60]
Affect Primary Monitor only: On
Affects second screen when changing to the Windows Vibrance Level which as i understand is the "Default" level to go back to when not affected by what program is running.
I have tried to beta version and its the same thing, can this program not change the values at all for the secondary screen when this setting is turned on?

**Comment by juvlarN** [comment]
Hook me up on Twitter https://twitter.com/juvlarN/ so I can direct message you. I might have a fix for this in mind.

**Comment by juv** [comment]
@DetectiveSquirrel any update?

**Comment by ajohnsen** [comment]
I have this problem as well.
It often affects my 2nd monitor and I have to reset the vibrance for this monitor manually.

My main monitor is 1920x1080 using DisplayPort while my 2nd monitor is 1920x1200 using DVI if that makes a difference.

**Comment by ajohnsen** [comment]
Nevermind, today i've had the same problem again.

Vibrance is still applied to secondary monitor after exiting game, even if i have said to only affect primary monitor.

**Comment by juv** [comment]
@ajohnsen can you explain more in depth what your initial setup is and when the change of vibrancy changes on your second monitor.

vibrancegui settings:
windows level: ? 
ingame level: ?

Which steps do you do?
1. start game, everything is fine (second monitor is unaffected but primary monitor is affected)?
2. tab out, everything is fine?
3. close game, second monitor is affected?

is this reproducible, e.g. it happens every time in the same way? 

and have you checked what your nvidia control panel reports as the Digital Vibrance value for your second monitor when you get it in that "unwanted" state? what number does it report?

**Comment by ajohnsen** [comment]
My settings are:
Windows vibrance: 50%
csgo vibrance: 100%

'affect primary monitor only' and 'never change resolution' is on

Sometime it seems like when i tab back and forth that its only applied to the 2nd monitor and never reset it.

I tried to reproduce right now, i had a youtube video in fullscreen on the 2nd monitor while having cs on the other. 
When i tabbed out of cs, maximized the youtube video, clicked a different program on the first monitor and then went back into cs it changed the vibrance on my 2nd monitor and not my first one. 
But when i tabbed out of cs, it fixed my 2nd monitor.

There has to be some kind of edge case, or a race condition or something. Because its very hard to actually reproduce.

Does it have any log files anywhere?

**Comment by juv** [comment]
There are no log files for tracking this down at the moment.
But it could be that Windows is assigning the game to your second monitor for a short amount of time when you tab out and because of the event driven nature of vibrancegui - as in "all of your application switches into foreground are given to vibrancegui by a windows api " - it is possible that the game is triggering a change for your second monitor. And after that switch, vibrancegui will then not retrigger because of the "affect primary monitor only" setting :/~

**Comment by juv** [comment]
@ajohnsen can you try this version here and see if this problem was fixed? https://github.com/juv/vibranceGUI/issues/95#issuecomment-750918557

**Comment by ajohnsen** [comment]
Tried it a little, and so far its working. I'll comment again if I see it happening :)

**Comment by ajohnsen** [comment]
Seems like this version also has this problem, I guess it happens when I alt tab a lot. Do you have a version that have some logging that i can monitor to see if i can find what triggers it?

**Comment by Manue3pii** [comment]
Thank you, starting csgo without "Affect primary monitor only" was the fix for me ^^

**Comment by vphcat** [comment]
Yes starting csgo without "Affect primary monitor only" fixed it for me aswell thank you :)

**Comment by juv** [comment]
Hi all,

can you please take a look at this new beta version i have built based on this PR: #140 
I tried to fix this issue.

You can download the pre-release .exe here:
https://github.com/juv/vibranceGUI/releases/tag/v2.5.0
